### PR TITLE
polish(errors): better state tree errors

### PR DESCRIPTION
**Pull request by frrist**
StateTree is failing to load for Lily is processing the chain. Make these errors more informative.

### CODE REVIEW

File: chain/state/statetree.go [hunk 1]
```diff
@@ -273,7 +273,7 @@ func LoadStateTree(cst cbor.IpldStore, c cid.Cid) (*StateTree, error) {
 	}
 	if err != nil {
 		log.Errorf("failed to load state tree: %s", err)
-		return nil, xerrors.Errorf("failed to load state tree: %w", err)
+		return nil, xerrors.Errorf("failed to load state tree %s: %w", c.String(), err)
```
Comment (Stebalien) [inline chain/state/statetree.go:276]: Don't bother calling `.String()`, it'll be called automatically when formatting.

**Review by Stebalien — APPROVED** [review]
LGTM modulo a nit.

**Review by whyrusleeping — APPROVED** [review]
what steb said

**Comment by frrist** [comment]
addressed.

**Review by Stebalien — APPROVED** [review]

**Pull request merged**